OMCompiler/Compiler/Template/CodegenUtil.tpl

```diff
@@ -168,12 +168,19 @@ template escapeCComments(String stringWithCComments)
 ::= '<%System.stringReplace(System.stringReplace(stringWithCComments, "/*", "(*"), "*/", "*)")%>'
 end escapeCComments;
 
-/*********************************************************/
-
-
-
-
+template crefCComment(SimVar v)
+"write the C comment for a cref, if it is not to be obfuscated"
+::=
+  match v
+  case SIMVAR(isProtected = true) then
+    if not boolOr(stringEq(getConfigString(OBFUSCATE), "protected"), stringEq(getConfigString(OBFUSCATE), "full"))
```

Comment:
@perost I don't know how to identify variables in encrypted models.

Reply:
This is just a C comment with the name of the variable, I don't think the frontend can remove that...

Thanks for the function pointer :slightly_smiling_face:  I can't really test it because I don't know how to build encrypted model parts, but it should work approximately correct, `none` and `full` do what they should at least.